backend/cfg/cfg_with_layout.mli

```diff
@@ -53,6 +53,8 @@ val assign_blocks_to_section : t -> Label.t list -> string -> unit
 
 val get_section : t -> Label.t -> string option
 
+val merge_section_with : t -> Label.t -> string option -> unit
```

Comment:
I would prefer `merge_section` that is symmetric and takes two labels, or rename this function to `set_section_name_option`.